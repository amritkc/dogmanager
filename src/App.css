body {
  font-family: Arial, sans-serif;
  background-color: #f5f5f5;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}
.App {
  text-align: center;
  padding: 20px;
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  max-width: 600px;
  width: 100%;
}
form {
  margin-bottom: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

input[type="text"],
input[type="number"] {
  padding: 10px;
  margin: 5px;
  border: 1px solid #ccc;
  border-radius: 4px;
  width: calc(100% - 22px);
  max-width: 300px;
}
button {
  padding: 10px 20px;
  margin: 5px;
  border: none;
  border-radius: 4px;
  background-color: #4CAF50;
  color: white;
  cursor: pointer;
}
button:hover {
  background-color: #45a049;
}

h2 {
  color: #333;
}

ul {
  list-style-type: none;
  padding: 0;
}

li {
  padding: 8px;
  margin: 5px 0;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 4px;
  text-align: left;
}

p {
  color: #666;
}