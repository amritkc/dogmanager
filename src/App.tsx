import React, { useState } from 'react';
import './App.css';
import DogList from './components/dogsList/dogList';
import SendDataButton from './components/sendDataButton/sendDataButton';
import { getCurrentDateTime } from './utils/getCurrentDate';
import DogForm from './components/dogForm/dogForm';
import { Dog } from './utils/constant';

function App() {
  const [dogs, setDogs] = useState<Dog[]>([]);
  const [lastUpdate, setLastUpdate] = useState('');

  const addDogs = (name: string, age: number) => {
    setDogs([...dogs, { name, age }])
    setLastUpdate(getCurrentDateTime())
  }

  return (
    <div className="App">
      <h1>Dog Manager</h1>
      <DogForm addDog={addDogs} />
      <DogList dogs={dogs} LastUpdate={lastUpdate} />
      <SendDataButton dogs={dogs} />
    </div>
  );
}

export default App;
