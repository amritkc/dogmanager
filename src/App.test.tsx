
import { render, screen } from '@testing-library/react';
import App from './App';

describe("App Test", () => {
  test("Validate App Header", () => {
    const dogManagerLabel = 'Dog Manager';
    render(<App />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toBe(dogManagerLabel);
  })


  test("Valadidate app Snapshot test", () => {
    const component = render(<App />);
    expect(component).toMatchSnapshot();
  })
})
