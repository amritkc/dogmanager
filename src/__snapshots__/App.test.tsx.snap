// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test SnapShot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1>
          Dog Manager
        </h1>
        <form>
          <input
            data-testid="dogsname"
            placeholder="Dog's Name"
            type="text"
            value=""
          />
          <input
            data-testid="dogsage"
            placeholder="Dog's Age"
            type="number"
            value=""
          />
          <button
            data-testid="submitdogslist"
            type="submit"
          >
            Submit
          </button>
        </form>
        <div>
          <h2>
            Dog's List
          </h2>
          <ul />
          <p>
            Last List Updated time: 
            never
          </p>
        </div>
        <button
          data-testid="send-button"
        >
          Send Data to Backend
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1>
        Dog Manager
      </h1>
      <form>
        <input
          data-testid="dogsname"
          placeholder="Dog's Name"
          type="text"
          value=""
        />
        <input
          data-testid="dogsage"
          placeholder="Dog's Age"
          type="number"
          value=""
        />
        <button
          data-testid="submitdogslist"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div>
        <h2>
          Dog's List
        </h2>
        <ul />
        <p>
          Last List Updated time: 
          never
        </p>
      </div>
      <button
        data-testid="send-button"
      >
        Send Data to Backend
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App Test Valadidate app SnapShot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1>
          Dog Manager
        </h1>
        <form>
          <input
            data-testid="dogsname"
            placeholder="Dog's Name"
            type="text"
            value=""
          />
          <input
            data-testid="dogsage"
            placeholder="Dog's Age"
            type="number"
            value=""
          />
          <button
            data-testid="submitdogslist"
            type="submit"
          >
            Submit
          </button>
        </form>
        <div>
          <h2>
            Dog's List
          </h2>
          <ul />
          <p>
            Last List Updated time: 
            never
          </p>
        </div>
        <button
          data-testid="send-button"
        >
          Send Data to Backend
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1>
        Dog Manager
      </h1>
      <form>
        <input
          data-testid="dogsname"
          placeholder="Dog's Name"
          type="text"
          value=""
        />
        <input
          data-testid="dogsage"
          placeholder="Dog's Age"
          type="number"
          value=""
        />
        <button
          data-testid="submitdogslist"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div>
        <h2>
          Dog's List
        </h2>
        <ul />
        <p>
          Last List Updated time: 
          never
        </p>
      </div>
      <button
        data-testid="send-button"
      >
        Send Data to Backend
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App Test Valadidate app Snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1>
          Dog Manager
        </h1>
        <form>
          <input
            data-testid="dogsname"
            placeholder="Dog's Name"
            type="text"
            value=""
          />
          <input
            data-testid="dogsage"
            placeholder="Dog's Age"
            type="number"
            value=""
          />
          <button
            data-testid="submitdogslist"
            type="submit"
          >
            Submit
          </button>
        </form>
        <div>
          <h2>
            Dog's List
          </h2>
          <ul />
          <p>
            Last List Updated time: 
            never
          </p>
        </div>
        <button
          data-testid="send-button"
        >
          Send Data to Backend
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1>
        Dog Manager
      </h1>
      <form>
        <input
          data-testid="dogsname"
          placeholder="Dog's Name"
          type="text"
          value=""
        />
        <input
          data-testid="dogsage"
          placeholder="Dog's Age"
          type="number"
          value=""
        />
        <button
          data-testid="submitdogslist"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div>
        <h2>
          Dog's List
        </h2>
        <ul />
        <p>
          Last List Updated time: 
          never
        </p>
      </div>
      <button
        data-testid="send-button"
      >
        Send Data to Backend
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
